goflags.go

```diff
@@ -32,103 +32,111 @@ type flagData struct {
 	defaultValue interface{}
 }
 
+// NewFlagSet creates a new flagSet structure for the application
+func NewFlagSet() *FlagSet {
+	return &FlagSet{flagKeys: *newInsertionOrderedMap()}
+}
+
+func newInsertionOrderedMap() *InsertionOrderedMap {
+	return &InsertionOrderedMap{
+		values: make(map[string]*flagData),
+		keys:   make([]string, 0, 0),
+	}
+}
+
 // Hash returns the unique hash for a flagData structure
 // NOTE: Hash panics when the structure cannot be hashed.
-func (f *flagData) Hash() string {
-	hash, _ := structhash.Hash(f, 1)
+func (flagSet *flagData) Hash() string {
+	hash, _ := structhash.Hash(flagSet, 1)
 	return hash
 }
 
-// New creates a new flagset structure for the application
-func New() *FlagSet {
```

Comment:
This `New()` Function should be kept as is for backward compatibility. Maybe we should just call `NewFlagSet()` internally for previous New function.

Reply:
Normally, if it would be a more complicated method with a lot of dependents, I'd mark the method as deprecated and would mention the intended replacement so that others would know how to make their code compliant/future-proof. I went with the direct change because it is a one word fix in every project that is using it.

Reply:
I assume the dependent projects use this library with a tagged version instead of the latest "snapshot", hence manual version incrementation is needed for upgrade = no project will be "automatically" broken?

Reply:
Sure, this also makes more sense from a naming perspective.